addons/account/models/chart_template.py

```diff
@@ -606,6 +610,158 @@ def _setup_utility_bank_accounts(self, template_code, company, template_data):
+        # - Create accounts at most probable location in the CoA and assign those to fields requiring an account
+        # - Creates tax group and taxes with their ir.model.data
+
+        taxes_in_country = self.env['account.tax'].search([
+            ('country_id', '=', country.id),
+            ('company_id', '=', company.id)
+        ])
+        if taxes_in_country:
+            return
+
+        def create_foreign_tax_account(existing_account, additional_label):
+            new_code = self.env['account.account']._search_new_account_code(
+                existing_account.company_id,
+                len(existing_account.code),
+                existing_account.code[:-2]
+            )
+            return self.env['account.account'].create({
+                'name': f"{existing_account.name} - {additional_label}",
+                'code': new_code,
+                'account_type': existing_account.account_type,
+                'company_id': existing_account.company_id.id,
+            })
+
+        existing_accounts = {'': None, None: None}  # keeps tracks of the created account by foreign xml_id
+        default_company_taxes = company.account_sale_tax_id + company.account_purchase_tax_id
+
+        def corresponding_tax_repartition_line_account(type_tax_use, foreign_tax_rep_line):
+            account_template_xml_id = foreign_tax_rep_line['account_id']
+            if account_template_xml_id in existing_accounts:
+                return existing_accounts[account_template_xml_id]
+
+            sign_comparator = '<' if float(foreign_tax_rep_line.get('factor_percent', 100)) < 0 else '>'
+            minimal_domain = [
+                ('company_id', '=', company.id),
+                ('account_id', '!=', False),
+                ('factor_percent', sign_comparator, 0),
+            ]
+            additional_domain = [
+                ('tax_id.type_tax_use', '=', type_tax_use),
+                ('tax_id.country_id', '=', company.account_fiscal_country_id.id),
+                ('tax_id', 'in', default_company_taxes.ids),
+            ]
+
+            # Trying to find an account being less restrictive on each iteration until the minimum acceptable is reached
+            # If nothing is found, don't fill it to avoid setting a wrong account
+            similar_repartition_line = None
+            while not similar_repartition_line and additional_domain:
+                search_domain = minimal_domain + additional_domain
+                similar_repartition_line = self.env['account.tax.repartition.line'].search(search_domain, limit=1)
+                additional_domain.pop()
+
+            similar_account_id = None
+            if similar_repartition_line:
+                local_tax_account = similar_repartition_line.account_id
+                similar_account_id = create_foreign_tax_account(local_tax_account, _("Foreign tax account (%s)", country.code)).id
+                existing_accounts[account_template_xml_id] = similar_account_id
```

Comment:
```suggestion
            existing_accounts[account_template_xml_id] = similar_account_id
```

Reply:
So you think I should build the mapping first and only then trigger the `create`?